lib/handlers/connect.js

```diff
@@ -75,7 +75,6 @@ function init (client, packet, done) {
     doConnack(
       { client: client, returnCode: returnCode, sessionPresent: false },
       done.bind(this, error))
-    client.conn.end()
```

Comment:
When will the client be disconnected then?

Reply:
Like I said in previous comment: doConnack will call done with `error`, this will be catched by client (nextBatch)[https://github.com/moscajs/aedes/blob/master/lib/client.js#L62] that will emit error that will call (client.onError)[https://github.com/moscajs/aedes/blob/master/lib/client.js#L92] that will close client connection [here](https://github.com/moscajs/aedes/blob/master/lib/client.js#L192)